googleapiclient/discovery.py

```diff
@@ -797,9 +800,17 @@ def method(self, **kwargs):
           msgRoot.attach(msg)
           # encode the body: note that we can't use `as_string`, because
           # it plays games with `From ` lines.
-          fp = StringIO()
-          g = Generator(fp, mangle_from_=False)
+          fp = BytesIO()
+          g = BytesGenerator(fp, mangle_from_=False)
+          # monkeypatch _write_lines to avoid munging '\r' into '\n'
+          # ( https://bugs.python.org/issue18886 );
+          # since Python 2 does not have _write_lines so this is a no-op
+          write_lines = getattr(BytesGenerator, "_write_lines", None)
+          if write_lines:
+            BytesGenerator._write_lines = BytesGenerator.write
           g.flatten(msgRoot, unixfrom=False)
```

Comment:
What happens if g.flatten throws an Exception? I assume BytesGenerator.write will still be patched.